crates/model/src/order.rs

```diff
@@ -564,6 +564,17 @@ impl OrderKind {
     }
 }
 
+impl TryFrom<[u8; 32]> for OrderKind {
+    type Error = anyhow::Error;
+    fn try_from(kind: [u8; 32]) -> Result<Self, Self::Error> {
+        match kind {
+            OrderData::KIND_SELL => Ok(OrderKind::Sell),
+            OrderData::KIND_BUY => Ok(OrderKind::Buy),
+            _ => Err(anyhow!("Order kind is not well defined")),
+        }
+    }
+}
```

Comment:
This would be better as a custom function so it is clear what kind of data the bytes are. Something like `from_contract_constant`. Same for the following functions.